NotesApp/Reusable/CustomAlertView.swift

```diff
@@ -0,0 +1,130 @@
+//
+//  CustomAlertView.swift
+//  NotesApp
+//
+//  Created by Luka Gujejiani on 31.10.24.
+//
+
+import UIKit
+
+final class CustomAlertView: UIView {
+    // MARK: - Properties
+    var onCancel: (() -> Void)?
+    var onDelete: (() -> Void)?
+    
+    private lazy var containerView: UIView = {
+        let view = UIView()
+        view.backgroundColor = UIColor(hex: "#252525")
+        view.layer.cornerRadius = 12
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+    
+    private lazy var infoImageView: UIImageView = {
+        let imageView = UIImageView()
+        let symbolConfig = UIImage.SymbolConfiguration(pointSize: 50, weight: .regular)
+        imageView.image = UIImage(systemName: "info.circle.fill", withConfiguration: symbolConfig)
+        imageView.tintColor = UIColor(hex: "#606060")
+        imageView.contentMode = .scaleAspectFit
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        return imageView
+    }()
+    
+    private lazy var messageLabel: UILabel = {
+        let label = UILabel()
+        label.text = "Are you sure you want to\ndelete this note?"
+        label.font = UIFont(name: "Nunito-Regular", size: 20)
+        label.textColor = UIColor(hex: "#CFCFCF")
+        label.numberOfLines = 0
+        label.textAlignment = .center
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    private lazy var cancelButton: UIButton = {
+        let button = UIButton(type: .system)
+        button.setTitle("Cancel", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.backgroundColor = UIColor(hex: "#30BE71")
+        button.layer.cornerRadius = 6
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addAction(UIAction(handler: { [weak self] _ in
+            self?.cancelButtonTapped()
+        }), for: .touchUpInside)
+        return button
+    }()
+    
+    private lazy var deleteButton: UIButton = {
+        let button = UIButton(type: .system)
+        button.setTitle("Delete", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.backgroundColor = UIColor(hex: "#FF0000")
+        button.layer.cornerRadius = 6
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addAction(UIAction(handler: { [weak self] _ in
+            self?.deleteButtonTapped()
+        }), for: .touchUpInside)
```

Comment:
Good that you used closure for action. addTarget(_:action:for:) approach is mostly used. why do you prefer closure rather that selector in this case?

Reply:
Using addAction with closures is a more modern approach that supports memory-safe weak references, improves readability, and avoids @objc selectors.